modules/400-descheduler/hooks/internal/api/v1alpha1/descheduler.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2021 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Descheduler is a description of a single descheduler instance
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:printcolumn:JSONPath=status.ready,name=Ready,type=boolean
+type Descheduler struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a descheduler instance.
+	Spec DeschedulerSpec `json:"spec"`
+
+	// Most recently observed status of a descheduler instance.
+	Status DeschedulerStatus `json:"status,omitempty"`
+}
+
+type DeschedulerSpec struct {
+	// Defines Template of a descedhuler Deployment
+	DeploymentTemplate DeschedulerDeploymentTemplate `json:"deploymentTemplate,omitempty"`
+
+	// commonParameters and strategies follow descheduler's documentation
+	// https://github.com/kubernetes-sigs/descheduler#policy-and-strategies
+	DeschedulerPolicy DeschedulerPolicy `json:"deschedulerPolicy,omitempty"`
+}
+
+type DeschedulerDeploymentTemplate struct {
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// If specified, the pod's tolerations.
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Resource allocation policy
+	ResourcesRequests *ResourcesRequests `json:"resourcesRequests,omitempty"`
+}
+
+type ResourcesRequests struct {
+	// Defines VPA mode that applies to a given descheduler Deployment
+	// +kubebuilder:default=VPA
+	// +kubebuilder:validation:Enum=Manual;VPA;Off
+	Mode ResourcesRequestsMode `json:"mode,omitempty"`
+}
+
+type ResourcesRequestsMode string
+
+const (
+	ResourcesRequestsModeManual = "Manual"
```

Comment:
These are strings. Is it ok? I assume from the code that they should have the `ResourcesRequestsMode` type.

Reply:
VPAs are no longer configured.